GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs

```diff
@@ -0,0 +1,167 @@
+                {
+                    continue;
+                }
+
+                foreach (var type in assembly.GetTypes())
+                {
+                    // Look for AssetBundleBrowserMain in the AssetBundleBrowser Namespace
+                    if (type != null && string.Equals(type.Namespace, "AssetBundleBrowser") &&
+                        type.Name.Equals("AssetBundleBrowserMain"))
+                    {
+                        _assetBundleBrowserIsPresent = true;
+                        return _assetBundleBrowserIsPresent.Value;
+                    }
+                }
+            }
+
+            _assetBundleBrowserIsPresent = false;
+            return false;
+        }
+
+        // Evaluates whether a folder has "AssetBundles-Browser" in its name, making it
+        // a candidate for the Asset Bundles Browser Folder
+        private static bool IsAssetBundleBrowserFolder(string folderName)
+        {
+            var regex = new Regex(@"AssetBundles-Browser", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+            return regex.Matches(folderName).Count > 0;
+        }
+
+        // Extracts AssetBundleBrowser version name from the Asset Bundle Browser package.json file
+        private static string GetBrowserVersion(bool useCurrentValueIfPresent = true)
+        {
+            if (useCurrentValueIfPresent && _assetBundleBrowserVersion != null)
+            {
+                return _assetBundleBrowserVersion;
+            }
+
+            var assetsPath = Application.dataPath;
+            var assetBundleBrowserFolderPaths =
+                Directory.GetDirectories(assetsPath).ToArray().Where(IsAssetBundleBrowserFolder);
+
+            foreach (var folderPath in assetBundleBrowserFolderPaths)
+            {
+                var expectedPackageDotJsonPath = Path.Combine(folderPath, "package.json");
+                if (!File.Exists(expectedPackageDotJsonPath))
+                {
+                    continue;
+                }
+
+                var data = File.ReadAllText(expectedPackageDotJsonPath);
+                try
+                {
+                    var json = JsonUtility.FromJson<PackageDotJSsonContent>(data);
+
+                    if (string.Equals(json.name, AssetBundleBrowserName))
+                    {
+                        _assetBundleBrowserVersion = json.version;
+                        return _assetBundleBrowserVersion;
+                    }
+                }
+                catch (ArgumentException e)
+                {
+                    Debug.LogWarning(string.Format(
+                        "Unable to read Asset Bundle Browser version contents from {0}. \n {1}",
+                        expectedPackageDotJsonPath, e.Message));
+                }
+            }
+
+            _assetBundleBrowserVersion = "not found";
+            return _assetBundleBrowserVersion;
+        }
+
+        /// <summary>
+        /// Display the Asset Bundle Browser Window
```

Comment:
Display the Asset Bundle Browser window or log an error if not present.

Reply:
Resolved